AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_script_canvas.py

```diff
@@ -62,14 +59,27 @@ def create_new_entity_with_component(self, entity_name: str, sc_file_path: str,
         self.editor_entity = EditorEntity.create_editor_entity_at(position, entity_name)
         self.editor_entity.add_components([SCRIPT_CANVAS_UI])
 
-        self.script_canvas_component = self.editor_entity.get_components_of_type([SCRIPT_CANVAS_UI])[0]
-        self.script_canvas_component.set_component_property_value(SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH, sourcehandle)
+        self.script_canvas_components = self.editor_entity.get_components_of_type([SCRIPT_CANVAS_UI])
+        self.script_canvas_components[0].set_component_property_value(SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH, sourcehandle)
+
+    def set_component_graph_file(self, sc_file_path: str, component_index=0) -> None:
+        """
+        Function for updating the script canvas file being used by the component
+
+        param sc_file_path : the string path the new graph file on disk
+        param component_index: the index of the script canvas component (entities can have multiple script canvas components)
 
-    def add_component_to_entity(self, editor_entity: EditorEntity, sc_file_path: str,
-                                sc_component_index=0) -> None:
+        returns none
         """
-        Function for constructing the SCComponent object. This uses an existing entity and loads a new script canvas
-        file into the source handle value
+        from editor_python_test_tools.editor_component.editor_component_validation import (
+            validate_script_canvas_graph_file)
+
+        sourcehandle = scriptcanvas.SourceHandleFromPath(sc_file_path)
+        validate_script_canvas_graph_file(self.get_script_canvas_components, sourcehandle, component_index)
```

Comment:
Validators do not belong in the setters and breaks the pattern set up in the the other Editor Component Behaviors model.

We should not have diverging patterns of development in these libraries.

Reply:
I made an attempt at putting the validation calls in the test scripts. The setters now return some data that the client can use to call the validation functions